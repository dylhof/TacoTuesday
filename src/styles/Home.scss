a {
  text-decoration: none;
}

.randoTacoRefresh {
  height: 300px;
  min-width: 300px;
  @include divFlex(center, center, row);
  @include text($title, 30px, center);
}