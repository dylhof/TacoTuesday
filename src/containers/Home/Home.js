import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Grid, Card, List, ListItem, ListItemText, Typography } from '@material-ui/core'
import { withStyles } from '@material-ui/core/styles'
import uuid from 'uuid/v4';
import {Link} from 'react-router-dom';
import { fetchRandoTaco } from '../../thunks/fetchRandoTaco';

const styles = {
  card: {
    margin: '10px',
    padding: '10px',
    height: '250px',
    opacity: '.8'
  },
  list: {
    height: '200px',
    overflow: 'scroll',
    border: '2px solid black',
    margin: '10px',
    padding: '3px',
    borderRadius: '5px'
  },

}

export class Home extends Component {
  mapRandoTaco = () => {
    const { randoTaco, classes } = this.props
    const randoTacoKeys = Object.keys(randoTaco)
    const jsxItems = randoTacoKeys.map(tacoLayer => {
      return (
        <Grid item key={randoTaco[tacoLayer].slug} xs={12} sm={6} md={4} lg={2}>
          <Link to={`/taco/${randoTaco[tacoLayer].slug}`}>
            <Card className={classes.card}>
              <h3>{randoTaco[tacoLayer].name}</h3>
              <List dense={true} className={classes.list}>{this.mapRecipe(randoTaco[tacoLayer].recipe)}</List>
            </Card>
          </Link>
        </Grid>)
    })
    return jsxItems
  }

  mapRecipe = (recipeText) => {
    const splitRecipe = recipeText.split('\n')
    const jsxRecipeItems = splitRecipe.map(step => {
      if (!step.includes('=')) {
        return (
          <ListItem key={uuid()}>
            <ListItemText>
              {step}
            </ListItemText>
          </ListItem>)
      }
    })
    return jsxRecipeItems
  }

  render() {
    return (
      <Grid container >
        {this.mapRandoTaco()}
        <Grid item xs={12} sm={6} md={4} lg={2} >
          <Card className={this.props.classes.card} onClick={this.props.fetchRandoTaco}> 
            <Typography>Click here To get a new Rando Taco!</Typography>
          </Card>
        </Grid>
      </Grid >
    )
  }
}

export const mapStateToProps = state => ({
  randoTaco: state.randoTaco
})

export const mapDispatchToProps = dispatch => ({
  fetchRandoTaco: () => dispatch(fetchRandoTaco())
})

export default withStyles(styles)(connect(mapStateToProps, mapDispatchToProps)(Home))