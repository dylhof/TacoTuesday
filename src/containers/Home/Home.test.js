import React from 'react';
import {shallow} from 'enzyme';
import { Home } from './Home';

describe('Home', () => {
  let wrapper
  let mockRandoTaco

  beforeEach(() => {
    mockRandoTaco = { baseLayer: { name: 'best taco', slug: 'best_taco', recipe: 'this is a great recipe \n it is yummy'}}
    wrapper = shallow(
      <Home randoTaco = {mockRandoTaco} />)
  })
  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
})