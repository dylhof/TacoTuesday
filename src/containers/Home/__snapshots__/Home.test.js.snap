// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should match the snapshot 1`] = `
<WithStyles(Grid)
  container={true}
>
  <WithStyles(Grid)
    item={true}
    key="6eb11400-4ecf-44ef-9b19-773029cb3f48"
    lg={3}
    md={4}
    sm={6}
    xs={12}
  >
    <Link
      replace={false}
      to="/best_taco"
    >
      <RecipeCard
        listStyleName="homeList"
        styleName="homeRecipes"
        tacoRecipe={
          Object {
            "name": "best taco",
            "recipe": "this is a great recipe 
 it is yummy",
            "slug": "best_taco",
          }
        }
      />
    </Link>
  </WithStyles(Grid)>
  <WithStyles(Grid)
    item={true}
    lg={2}
    md={4}
    sm={6}
    xs={12}
  >
    <div
      className="recipeCard randoTacoRefresh"
    >
      <h2>
        Click here To get a new Rando Taco!
      </h2>
    </div>
  </WithStyles(Grid)>
</WithStyles(Grid)>
`;
